scipy/spatial/distance.py

```diff
@@ -185,6 +194,11 @@ def _nbool_correspond_ft_tf(u, v):
     return (nft, ntf)
 
 
+def _row_norms(X):
```

Comment:
This is only used once, in the deprecated `old_cosine` implementation. I'd just inline it there so it gets cleaned up when that code path is removed.